service-daemon

```diff
@@ -1,47 +1,91 @@
-#!/data/data/com.termux/files/usr/bin/sh
+#!/usr/bin/env sh
 
-set -e
-start(){
-    mkdir -p $PIDDIR
-    start-stop-daemon -S -b -m -p $PIDFILE -x $DAEMON -d $PREFIX -- $DAEMON_OPTS
+set -eu
+
+_prefix="${TERMUX__PREFIX:-"${PREFIX}"}"
+
+: "${XDG_RUNTIME_DIR:="${_prefix}/var/run"}"
+export XDG_RUNTIME_DIR
+
+BINDIR="${_prefix}/bin"
+readonly BINDIR
+
+LOGDIR="${_prefix}/var/log"
+export LOGDIR
+readonly LOGDIR
+
+SVDIR="${_prefix}/var/service"
+export SVDIR
+readonly SVDIR
+
+unset -v _prefix
+
+
+wait_for_pid() (
+    local pid="$1"
+    local s
+
+    for s in 1 2 3 4 5
+    do
+        kill -0 "${pid}" 2>/dev/null || break
+        sleep "$s"
+    done
+    kill -0 "${pid}" 2>/dev/null && return 1 || return 0
+}
+
+kill_from_pid_file() {
+    local pid s
+
+    for pid in $(head -n 1 "$1")
+    do
+        kill "${pid}"
+        wait_for_pid "${pid}" || kill -9 "${pid}"
+    done
+}
+
+start() {
+    local prefix="${TERMUX__PREFIX:-"${PREFIX}"}"
+    readonly prefix
+
+    mkdir -m 00700 -p "${XDG_RUNTIME_DIR}"
+    start-stop-daemon -S -b -m -p "${PIDFILE}" -x "${DAEMON}" -d "${prefix}" -- ${DAEMON_OPTS}
 }
-stop(){
-    start-stop-daemon -K -s 1 -o -p $PIDFILE
-    tail -f --pid=`cat $PIDFILE` /dev/null
-    for f in $SVDIR/*/supervise $SVDIR/*/log/supervise;do
-        kill -9 `cat $f/pid` 2>/dev/null ||:
-        rm -rf $f
+
+stop() {
+    local d pid="$(cat "${PIDFILE}")"
+
+    start-stop-daemon -K -s 1 -o -p "${PIDFILE}"
+    wait_for_pid "${pid}" || tail -f --pid="${pid}" /dev/null
+    for d in "${SVDIR}"/*/supervise "${SVDIR}"/*/log/supervise
```

Comment:
- `$d` is unnecessarily terse, consider a more descriptive name like `$service` or even just `$dir`.
- We usually put the `do` on the same line as the `for` or `while`.
- You can consolidate this to a single glob.
```suggestion
    for service in "${SVDIR}"/*/{,log/}supervise; do
```